lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1368,6 +1296,72 @@ def set_slug(
         self.service.shareable_service.set_slug(trans, id, payload)
         return Response(status_code=status.HTTP_204_NO_CONTENT)
 
+    @router.delete(
+        "/api/workflows/{workflow_id}",
+        summary="Add the deleted flag to a workflow.",
+    )
+    def delete_workflow(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+    ):
+        self.service.delete(trans, workflow_id)
+        return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+    @router.post(
+        "/api/workflows/{workflow_id}/undelete",
+        summary="Remove the deleted flag from a workflow.",
+    )
+    def undelete_workflow(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+    ):
+        self.service.undelete(trans, workflow_id)
+        return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+    @router.get(
+        "/api/workflows/{workflow_id}/versions",
+        summary="List all versions of a workflow.",
+    )
+    def show_versions(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+        instance: Optional[bool] = InstanceQueryParam,
+    ):
+        return self.service.get_versions(trans, workflow_id, instance)
+
+    @router.get(
+        "/api/workflows/menu",
+        summary="Get workflows present in the tools panel.",
+    )
+    def get_workflow_menu(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        show_deleted: Optional[bool] = DeletedQueryParam,
+        show_hidden: Optional[bool] = HiddenQueryParam,
+        missing_tools: Optional[bool] = MissingToolsQueryParam,
+        show_published: Optional[bool] = ShowPublishedQueryParam,
+        show_shared: Optional[bool] = ShowSharedQueryParam,
+    ):
+        payload = WorkflowIndexPayload(
```

Comment:
@davelopez is this the preferred way to move params around?

Reply:
Yes, that looks ok. You can also skip the "redundant" pydantic validation using:
```suggestion
        payload = WorkflowIndexPayload.construct(
```
since your query parameters were already validated by FastAPI and you know they must be correct. See https://pydantic-docs.helpmanual.io/usage/models/#creating-models-without-validation for details. But maybe this is just an overoptimization :sweat_smile: